src/main/java/org/reactome/qa/doisuggester/ReactionlikeEvent.java

```diff
@@ -0,0 +1,292 @@
+		}
+		return this.ancestors;
+	}
+
+	/**
+	 * Private recursive method to search up the Pathway hierarchy.
+	 * This method gets the ancestors of a pathway/RLE that is furthest up the tree and also does not have the same InstanceEdit.
+	 * @param A pathway to examine.
+	 * @param newInstanceEdit the InstanceEdit to check for.
+	 * @return The pathway objects.
+	 */
+	private Set<GKInstance> getFurthestAncestorWithoutNewInstanceEdit(GKInstance pathway, GKInstance newInstanceEdit)
+	{
+		try
+		{
+			List<GKInstance> parents = (List<GKInstance>) pathway.getReferers(ReactomeJavaConstants.hasEvent);
+			// Only proceed if there are valid parents. If there are no valid parents, then don't do anything because
+			// this process only makes sense if pathway has parents.
+			if (! (parents == null || parents.isEmpty()) )
+			{
+				// There are a few ReactionlikeEvents that are referred to by multiple Pathways via hasEvent...
+				// For example, R-HSA-349426 ("Phosphorylation of MDM4 by CHEK2") is in both "Stabilization of p53"
+				// under the top-level pathway Cell Cycle, and "Regulation of TP53 Degradation" under the top-level
+				// pathway Gene Expression.
+				// Both of these parents must be recursively checked.
+				for (GKInstance parent : parents)
+				{
+					List<GKInstance> parentInstanceEdits = new ArrayList<>();
+					// Ok, technically, the parent object should be a Pathway, and a ReactionlikeEvent
+					// is *not* a subclass of Pathway, but they are both subclasses of Event,
+					// and Event has the attributes "authored", "revised", "reviewed", so should work.
+					ReactionlikeEvent parentPathway = new ReactionlikeEvent(parent);
+					parentInstanceEdits.addAll(parentPathway.getAuthoredInstanceEdits());
+					parentInstanceEdits.addAll(parentPathway.getReviewedInstanceEdits());
+					parentInstanceEdits.addAll(parentPathway.getRevisedInstanceEdits());
+					boolean instanceEditFound = false;
+					int i = 0;
+					while (!instanceEditFound && i < parentInstanceEdits.size())
+					{
+						GKInstance parentInstanceEdit = parentInstanceEdits.get(i);
+						if (newInstanceEdit.getDBID().equals(parentInstanceEdit.getDBID()))
+						{
+							instanceEditFound = true;
+							// If an ancestor of the RLE has the same InstanceEdit, it means we need to go up one more level to find an ancestor
+							// that does *not* have the InstanceEdit
+							this.ancestors.addAll(this.getFurthestAncestorWithoutNewInstanceEdit(parent, newInstanceEdit));
+						}
+						i++;
+					}
+
+					if (!instanceEditFound)
+					{
+						// A pathway that does not have the "new" instanceEdit is the sort of pathway we're interested in
+						// so add it to the list of ancestors that will be returned.
+						// If pathway != this.rle, add pathway, otherwise, add parent.
+						if (pathway.getDBID().equals(this.rle.getDBID()))
+						{
+							this.ancestors.add(parent);
+						}
+						else
+						{
+							this.ancestors.add(pathway);
+
+						}
```

Comment:
I'm a bit confused by this check.  Is it the presence or absence of the new instance edit that determines whether the parent of the pathway or the itself the pathway should be added?

If so, should the if condition be different because it is checking if the pathway is actually the underlying reaction like event which won't be applicable once the method recurses and `pathways` is a pathway and its `parent` is a parent pathway.

Reply:
Yeah, it is a bit confusing. I was confused too, for a while. ;)

Well, at first I was always adding `parent`. But I noticed that for some pathways, there were inconsistent results when compared to Justin's output, which had already been verified by Lisa. When I always added `pathway`, there were different inconsistencies. 

A better explanation in words:

If `pathway` is the same entity as the underlying RLE _and_ the new InstanceEdit was not found in any ancestors, then `parent` is an ancestor for `pathway` that is unmodified, and should go into the resultset. 

BUT... if `pathway` is NOT the underlying RLE and the new InstanceEdit was not found amongst the ancestors, then we are already on an ancestor of the underlying RLE, so it is `pathway` that must be added to the resultset.

Does this make more sense?

Reply:
Okay, I see.  Could the "pathway" variable be renamed to "event" and the ReactionlikeEvent class be renamed to Event also to reflect it can describe a pathway or RlE?

Reply:
Ok I'll look into variable renaming.

Reply:
After thinking about it, I'd prefer to _not_ rename the class. The program specifically looks at ReactionlikeEvents. In the main method, the initial retrieval of objects to process looks like this:

```java
Collection<GKInstance> reactionLikeEvents = (Collection<GKInstance>) dbAdaptor.fetchInstancesByClass(ReactomeJavaConstants.ReactionlikeEvent);
```

I think renaming it could potentially cause confusion. If the scope of the program expands to cover other event-types then I think renaming the `ReactionlikeEvent` class to `Event` would make sense, but I don't think so right now.

Reply:
True, but the class is used to refer to pathways also in the ReactionlikeEvent class on lines 242-248.

Reply:
I think the fact that the whole project is about checking _ReactionlikeEvent_ takes priority, in this caes. And there is a comment explaining the situation, so I don't think there should be any problem.

Reply:
Okay, I won't insist, but I do think since the structure could be a Pathway it could be misleading for the class to be named ReactionlikeEvent.

Again, I won't insist, but it might make sense to have an Event abstract class or interface and then have ReactionlikeEvent and Pathway classes to represent each possibility so you could still use the ReactionlikeEvent class when needed without having to represent a Pathway as ReactionlikeEvent.  If you decide it's worth it, maybe for a future PR but I won't block this one.